Создание коллекции объектов

NBronina opened the pull request:
Не знаю, как объединить объекты в коллекцию, сделала через массив. Подскажите, как исправить. Еще я не понимаю, как эти объекты связать с объектом Event, который был реализован во 2ом задании, чтобы поля заданные по умолчанию в Event нашли отражение в создаваемых объектах.

NBronina committed:
Создание коллекции объектов

Удалось создать только массив объектов, а не коллекцию. Эти объекты
никак не связаны с типом объекта Event, который был реализован во 2ом
задании, соответственно у этих объектов нет значений заданных по
умолчанию. Нужно переделать.

azproduction commented:
Коллекция - просто массив из объектов типа Event. Все фильтры нужно делать над этим массивом.

NBronina committed:
Функции фильтрации и сортировки

Переделано создание коллекции объектов. Написан ряд функций сортировки и
фильтрации

NBronina commented:
Не понимаю, как работать со значения по умолчанию. Как мне создать объекта класса Event у которого бы поле name было заполнено значением по умолчанию.

azproduction commented:
Оператор `||` https://speakerdeck.com/azproduction/kripi-osnovy-javascript?slide=47

``` javascript
function a(optionalParameter) {
    optionalParameter = optionalParameter || "default value";
}
```
